### ISSUE TITLE
Presenting a swapchain image that has not been written to results in invalid API usage

### ISSUE THREAD

**Issue by dylanede** [tomaka/vulkano#519]
In other words, if you have an "empty" render loop that just presents images that have never been rendered to, you get error messages like this from the debug layers:

```text
MEM warning: vkQueuePresentKHR(): Cannot read invalid swapchain image 0x2775c9be030, please fill the memory before using.
DS error: Images passed to present must be in layout VK_IMAGE_LAYOUT_PRESENT_SRC_KHR or VK_IMAGE_LAYOUT_SHARED_PRESENT_KHR but is in VK_IMAGE_LAYOUT_UNDEFINED. For more information refer to Vulkan Spec Section '30.8. WSI Swapchain' which states 'Any given element of pImageIndices must be the index of a presentable image acquired from the swapchain specified by the corresponding element of the pSwapchains array, and the presented image subresource must be in the VK_IMAGE_LAYOUT_PRESENT_SRC_KHR or VK_IMAGE_LAYOUT_SHARED_PRESENT_KHR layout at the time the operation is executed on a VkDevice' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#VkPresentInfoKHR)
```

**Comment by dylanede** [comment]
I've also found that even when the images are drawn to, they still don't end up with the right layout at the point at which they're presented, instead having layout `VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL`.

**Comment by tomaka** [comment]
For the record we want to fix #497 first, then restore [this check](https://github.com/tomaka/vulkano/blob/934169ddfbf4c47779eea4e1e2f184de281abd4d/vulkano/src/swapchain/swapchain.rs#L123-L128).

While the "normal path" would work, you would get an error if you try to render to an image which you haven't drawn upon.
This is a bit problematic. On one hand we don't want to magically generate a command buffer that performs the transition (even though we could). On the other hand the error will be very confusing and difficult to understand.

**Comment by Rua** [comment]
I just ran into this issue. If I present the swapchain image immediately after acquiring it, without submitting any command buffers, I get the same validation errors.
